purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -134,12 +134,11 @@ class Purchases internal constructor(
      * Make sure [removeUpdatedCustomerInfoListener] is called when the listener needs to be destroyed.
      */
     var updatedCustomerInfoListener: UpdatedCustomerInfoListener?
-        @Synchronized get() = state.updatedCustomerInfoListener
+        @Synchronized get() = customerInfoHelper.updatedCustomerInfoListener
         set(value) {
             synchronized(this@Purchases) {
-                state = state.copy(updatedCustomerInfoListener = value)
+                customerInfoHelper.updatedCustomerInfoListener = value
```

Comment:
I made the listener on the `Purchases` class just a proxy for the one in the `CustomerInfoHelper`. Most operations related with customer info now will need to go through there.